overlord/fdestate/fdemgr_test.go

```diff
@@ -37,11 +42,126 @@ var _ = Suite(&fdeMgrSuite{})
+		return []byte{1, 2, 3, 4}, []byte{5, 6, 7, 8}, nil
+	})()
+
+	defer fdestate.MockVerifyPrimaryKeyHash(func(devicePath string, alg crypto.Hash, salt, digest []byte) (bool, error) {
+		c.Check(devicePath, Equals, "/dev/disk/by-uuid/bbb")
+		c.Check(alg, Equals, crypto.Hash(crypto.SHA256))
+		c.Check(salt, DeepEquals, []byte{1, 2, 3, 4})
+		c.Check(digest, DeepEquals, []byte{5, 6, 7, 8})
+		return true, nil
+	})()
+
 	runner := state.NewTaskRunner(st)
 	manager := fdestate.Manager(st, runner)
-	c.Assert(manager.Ensure(), IsNil)
+	c.Assert(manager.StartUp(), IsNil)
 	st.Lock()
 	defer st.Unlock()
 	foundManager := fdestate.FdeMgr(st)
 	c.Check(foundManager, Equals, manager)
+
+	var fdeSt fdestate.FdeState
+	err := st.Get("fde", &fdeSt)
+	c.Assert(err, IsNil)
+	primaryKey, hasPrimaryKey := fdeSt.PrimaryKeys[0]
+	c.Assert(hasPrimaryKey, Equals, true)
+	c.Check(crypto.Hash(primaryKey.Digest.Algorithm), Equals, crypto.Hash(crypto.SHA256))
+	c.Check(primaryKey.Digest.Salt, DeepEquals, []byte{1, 2, 3, 4})
+	c.Check(primaryKey.Digest.Digest, DeepEquals, []byte{5, 6, 7, 8})
+}
+
+type mockModel struct {
+}
+
+func (m *mockModel) Series() string {
+	return "mock-series"
+}
+
+func (m *mockModel) BrandID() string {
+	return "mock-brand"
+}
+
+func (m *mockModel) Model() string {
+	return "mock-model"
+}
+
+func (m *mockModel) Classic() bool {
+	return false
+}
+
+func (m *mockModel) Grade() asserts.ModelGrade {
+	return asserts.ModelSigned
+}
+
+func (m *mockModel) SignKeyID() string {
+	return "mock-key"
+}
+
+func (s *fdeMgrSuite) TestUpdateState(c *C) {
+	st := state.New(nil)
+
+	defer fdestate.MockDMCryptUUIDFromMountPoint(func(mountpoint string) (string, error) {
+		switch mountpoint {
+		case dirs.GlobalRootDir:
+			return "aaa", nil
+		case dirs.SnapSaveDir:
+			return "bbb", nil
+		}
+		return "", fmt.Errorf("unknown mount point")
+	})()
+
+	defer fdestate.MockGetPrimaryKeyHash(func(devicePath string, alg crypto.Hash) ([]byte, []byte, error) {
+		c.Check(devicePath, Equals, "/dev/disk/by-uuid/aaa")
+		c.Check(alg, Equals, crypto.Hash(crypto.SHA256))
+		return []byte{1, 2, 3, 4}, []byte{5, 6, 7, 8}, nil
+	})()
+
+	defer fdestate.MockVerifyPrimaryKeyHash(func(devicePath string, alg crypto.Hash, salt, digest []byte) (bool, error) {
+		c.Check(devicePath, Equals, "/dev/disk/by-uuid/bbb")
+		c.Check(alg, Equals, crypto.Hash(crypto.SHA256))
+		c.Check(salt, DeepEquals, []byte{1, 2, 3, 4})
+		c.Check(digest, DeepEquals, []byte{5, 6, 7, 8})
+		return true, nil
+	})()
+
+	runner := state.NewTaskRunner(st)
+	manager := fdestate.Manager(st, runner)
+	c.Assert(manager.StartUp(), IsNil)
+	st.Lock()
+	defer st.Unlock()
+	foundManager := fdestate.FdeMgr(st)
+	c.Check(foundManager, Equals, manager)
+
+	models := []secboot.ModelForSealing{
+		&mockModel{},
+	}
+
+	fdestate.UpdateParameters(st, "run+recover", "container-role", []string{"run"}, models, secboot.SerializedPCRProfile(`"serialized-profile"`))
+
+	var fdeSt fdestate.FdeState
+	err := st.Get("fde", &fdeSt)
```

Comment:
and to make sure serialization for SerialziedPCRProfile is working correctly I think I'd add something like:

```
var raw map[string]any
c.Assert(t.Get("fde", &raw), Nil)
c.Check(raw, DeepEquals, map[string]any{
	...
	"tpm2-prc-profile": map[string]any {
		"something": "serialized-profile",
	}
})
```

Reply:
I do not think that would work. The `TPM2PCRProfile` field is kept serialized.

Reply:
Ah I get it.

Reply:
I have put a test in `secboot` because this is where we define that type.

Reply:
thanks